lightning/src/ln/outbound_payment.rs

```diff
@@ -1827,6 +1838,25 @@ impl OutboundPayments {
 	pub fn clear_pending_payments(&self) {
 		self.pending_outbound_payments.lock().unwrap().clear()
 	}
+
+	pub fn get_invoice_request_awaiting_invoice(&self) -> Vec<(PaymentId, InvoiceRequest)> {
```

Comment:
Let's use `release` instead of `get` to note that successive calls will not return the same items.

Reply:
Done! [pr3010.17](https://github.com/shaavan/rust-lightning/commits/pr3010.17)
Thanks!